lib/pages/common_widgets/goal_widget.dart

```diff
@@ -123,6 +181,56 @@ class GoalWidget extends StatelessWidget {
             );
   }
 
+  Future<dynamic> selectExtension(BuildContext context, Null doExtendGoal(), goalExtensionInput) {
+    return showDialog(
+      context: context,
+      builder: (BuildContext context) => AlertDialog(
+        title: const Text('Enter how many extra days you need:'),
+        content: //const Text('Selection will be here'),
```

Comment:
There should be an alert dialog upon failure

Reply:
done